### PR TITLE
[video-player] upgraded video_player_interface to use pigeon

### PR DESCRIPTION
**Pull request by gaaclarke**
## Description

Upgraded to use pigeon, child commit: https://github.com/flutter/plugins/pull/2544/

## Related Issues

Related: flutter/flutter#32930

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [x] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [ ] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by gaaclarke** [comment]
@cyanglaz CI is failing because of a warning on breaking changes.  @amirh just said we have to ignore it if we want to submit it.  This is a breaking change because it changes the format of the messages, the actual API doesn't change at all.

**Comment by gaaclarke** [comment]
@cyanglaz Alternatively we could publish this as a whole different package if it makes getting through CI easier.  This could be `video_player_interface_pigeon`.

**Comment by xster** [comment]
LGTM

File: packages/video_player/video_player_platform_interface/lib/messages.dart [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+// See also: https://pub.dev/packages/pigeon
+// ignore_for_file: public_member_api_docs, non_constant_identifier_names, avoid_as, unused_import
+import 'dart:async';
+import 'package:flutter/services.dart';
+
+class TextureMessage {
+  int textureId;
+  // ignore: unused_element
+  Map<dynamic, dynamic> _toMap() {
+    final Map<dynamic, dynamic> pigeonMap = <dynamic, dynamic>{};
```
[the remark this answers is not in the document]
Comment (gaaclarke) [reply packages/video_player/video_player_platform_interface/lib/messages.dart:11]: I want to change it to Map<String, dynamic>, i just haven't gotten around to it.  I'm kind of stuck with what the standard codec supports.

**Comment by gaaclarke** [comment]
Landing red as is the protocol that was described to me by @amirh.  Look at https://github.com/flutter/plugins/pull/2617/commits/ecdeb4fc9821716bc463c90cb479e0f0dae08657 to see this passing most CI except the following:
1) Tests, those are expected to be broken until video_player is updated (which can't happen until this publishes).
2) Warning  about version prerelease versioning which we've always disregarded.
3) Warning about avoiding incrementing the major version, which we are doing here on purpose.

**Pull request merged**